### PR TITLE
Fix ITAB_DUPLICATE_KEY when using hashed table

### LINKED ISSUE #58: Dump ITAB_DUPLICATE_KEY when using hashed table
```abap
DATA itab TYPE HASHED TABLE OF i WITH UNIQUE KEY table_line.
itab = VALUE #( ( 3 ) ( 2 ) ( 1 ) ).

DATA lo_ajson TYPE REF TO zcl_ajson.

lo_ajson = zcl_ajson=>create_empty( ).
lo_ajson->keep_item_order( ).
lo_ajson->set(
  iv_path = '/'
  iv_val = itab ).
```

Loop over hashed table does *not* set sy-tabix.
https://github.com/sbcgua/ajson/blob/master/src/zcl_ajson.clas.locals_imp.abap#L1185

### PR DESCRIPTION
**Pull request by mbtools**
Closes #58

### CODE REVIEW

**Comment by sbcgua** [comment]
Nice fix, thanks !
Could you pls add a UT for it.

**Review by sbcgua — CHANGES_REQUESTED** [review]
Could you pls add a UT for the hashed table

**Comment by mbtools** [comment]
done

File: src/zcl_ajson.clas.testclasses.abap [hunk 1]
```diff
@@ -1931,6 +1932,35 @@ class ltcl_writer_test implementation.
...
+    data lo_nodes type ref to lcl_nodes_helper.
+    data lo_cut type ref to zcl_ajson.
+    data li_writer type ref to zif_ajson_writer.
+    data lt_tab type hashed table of string with unique default key.
+
+    lo_cut = zcl_ajson=>create_empty( ).
+    li_writer = lo_cut.
+
+    insert |hello| into table lt_tab.
+    insert |world| into table lt_tab.
```
Comment (sbcgua) [inline src/zcl_ajson.clas.testclasses.abap:1946]: image

731 does not support string templates inserts.

**Review by sbcgua — CHANGES_REQUESTED** [review]
Almost there :)

File: src/zcl_ajson.clas.testclasses.abap (hunk 1 above)
Comment (mbtools) [reply src/zcl_ajson.clas.testclasses.abap]: the usual... you make a change and go away fully believing you knew it works. well, it should now.

Comment (sbcgua) [reply src/zcl_ajson.clas.testclasses.abap]: I nearly missed it as well. The code looked good. And I nearly clicked merge button but then decided to test it :)

**Review by sbcgua — APPROVED** [review]
Now it's good, thanks ! Merging

**Pull request merged**